### PR TITLE
metadata as optional and validated

### PR DESCRIPTION
**Pull request by luiztauffer**
fix [#190](https://github.com/catalystneuro/nwb-conversion-tools/issues/190)

## Motivation
1. allow metadata to be optional for `Converter.run_conversion()`. If not passed by the user, Converter will use the basic metadata formed with `Converter.get_metadata()`
2. validate metadata before running conversion. This will help us catch metadata-related errors before conversion starts

## How to test the behavior?
All conversion tests will necessarily go through this

### CODE REVIEW

File: nwb_conversion_tools/nwbconverter.py [hunk 1]
```diff
@@ -143,15 +143,19 @@ def run_conversion(
         assert (not save_to_file and nwbfile_path is None) or nwbfile is None, \
             "Either pass a nwbfile_path location with save_to_file=True, or a nwbfile object, but not both!"
 
+        # Validate metadata
+        if metadata is None:
+            metadata = self.get_metadata()
+        else:
+            self.validate_metadata(metadata=metadata)
```
Comment (CodyCBakerPhD) [inline nwb_conversion_tools/nwbconverter.py:150]: @luiztauffer The key assumption of course being that everything from self..get_metadata() is ensured to pass validation? No edge cases where it could fail?

I know we have built-in tests for this that check each individual data interface metadata, but a converter can take deep unions of that, and I just wonder if that could ever have the potential to give a post-unification metadata that isn't valid?

Comment (luiztauffer) [reply nwb_conversion_tools/nwbconverter.py:150]: >@luiztauffer The key assumption of course being that everything from self..get_metadata() is ensured to pass validation? 

That's correct.

> but a converter can take deep unions of that, and I just wonder if that could ever have the potential to give a post-unification metadata that isn't valid?

in this case I believe the best thing to do would be override the run_conversion method of the edgy case converter
```
def run_conversion(…)
	# code that reorganizes metadata and make it schema compatible
	metadata = ….
	super().run_conversion(metadata, args...)
```

Comment (luiztauffer) [reply nwb_conversion_tools/nwbconverter.py:150]: or actually, just override the `Converter.get_metadata()` would do it.

Comment (CodyCBakerPhD) [reply nwb_conversion_tools/nwbconverter.py:150]: Seems more complicated than I was thinking - I was simply proposing that we remove the else and always validate metadata, regardless of if it was passed or auto-generated.

```suggestion
            metadata = self.get_metadata()
        self.validate_metadata(metadata=metadata)
```

Comment (luiztauffer) [reply nwb_conversion_tools/nwbconverter.py]: of course, that makes sense

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
I'll merge once the tests complete
